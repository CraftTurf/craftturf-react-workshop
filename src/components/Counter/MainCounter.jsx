import * as React from 'react';
import Counter from './Counter';

export default class MainCounter extends React.Component
{
  constructor(props) {
    super(props);

    this.state = {
      count: 0
    };

    this.handleClickDecrement = this.handleClickDecrement.bind(this);
    this.handleClickIncrement = this.handleClickIncrement.bind(this);
  }


  render() {
    return(
      <div>
        <p>{this.state.count}</p>
      <Counter onClick={this.handleClickIncrement}>
        {'Increment'}
      </Counter>

      <Counter onClick={this.handleClickDecrement}>
        {'Decrement'}
      </Counter>
    </div>
    );
  }

  handleClickIncrement() {
    const { count } = this.state;
    this.setState({
      count: count + 1
    })
  }

  handleClickDecrement() {
    const { count } = this.state;
    this.setState({
      count: count - 1
    })
  }
}

MainCounter.defaultProps = {
  children: 'Click The Button'
}
