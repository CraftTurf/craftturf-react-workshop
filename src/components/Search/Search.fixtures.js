/* eslint-disable */
const SearchFixtures = [
  {
    "id": "58861417c0f55e233f673bb2",
    "index": 0,
    "guid": "9a7f0af9-55f9-4e34-93e4-56a439111b62",
    "isActive": true,
    "balance": "$2,076.74",
    "picture": "http://placehold.it/32x32",
    "age": 21,
    "eyeColor": "brown",
    "name": "Kayla Foster",
    "gender": "female",
    "company": "ZENTURY",
    "email": "kaylafoster@zentury.com",
    "phone": "+1 (916) 502-3087",
    "address": "151 Ferry Place, Corriganville, Michigan, 3545",
    "about": "Nulla esse quis quis sint amet. Culpa ipsum consectetur enim aliquip. Sint enim cupidatat veniam dolore dolore minim. Nisi voluptate incididunt minim pariatur velit qui fugiat veniam consectetur officia voluptate. Lorem voluptate excepteur veniam deserunt cillum.\r\n",
    "registered": "2014-07-21T10:55:17 -01:00",
    "latitude": -24.160625,
    "longitude": -131.490855,
    "tags": [
      "non",
      "magna",
      "voluptate",
      "qui",
      "dolore",
      "consectetur",
      "Lorem"
    ],
    "greeting": "Hello, Kayla Foster! You have 10 unread messages.",
    "favoriteFruit": "strawberry"
  },
  {
    "id": "58861417b6e083669643ca32",
    "index": 1,
    "guid": "7478d2c8-7b71-491e-bc8e-f147a498f98e",
    "isActive": false,
    "balance": "$1,220.44",
    "picture": "http://placehold.it/32x32",
    "age": 26,
    "eyeColor": "brown",
    "name": "Jane Montgomery",
    "gender": "female",
    "company": "ULTRASURE",
    "email": "janemontgomery@ultrasure.com",
    "phone": "+1 (981) 596-2048",
    "address": "324 Canton Court, Yorklyn, Oregon, 6800",
    "about": "Non sint nostrud nostrud nisi duis deserunt aute sit consequat elit fugiat nulla quis. Sint esse voluptate proident nostrud excepteur ex ipsum ipsum exercitation proident et. Anim Lorem occaecat elit consectetur dolore sint adipisicing laboris voluptate cupidatat reprehenderit ullamco qui qui.\r\n",
    "registered": "2015-11-01T09:15:50 -01:00",
    "latitude": 64.3878,
    "longitude": 25.210197,
    "tags": [
      "id",
      "proident",
      "anim",
      "eiusmod",
      "incididunt",
      "reprehenderit",
      "incididunt"
    ],
    "greeting": "Hello, Jane Montgomery! You have 5 unread messages.",
    "favoriteFruit": "strawberry"
  },
  {
    "id": "588614176f0540806990e677",
    "index": 2,
    "guid": "0999988d-280c-4830-b1ab-74465e1e4386",
    "isActive": true,
    "balance": "$2,526.01",
    "picture": "http://placehold.it/32x32",
    "age": 38,
    "eyeColor": "blue",
    "name": "Kent Newton",
    "gender": "male",
    "company": "PYRAMI",
    "email": "kentnewton@pyrami.com",
    "phone": "+1 (843) 523-2642",
    "address": "849 Radde Place, Yukon, Idaho, 191",
    "about": "Do magna magna anim fugiat ullamco fugiat mollit aute magna irure ullamco exercitation. Do non pariatur deserunt deserunt anim veniam est ut labore reprehenderit culpa. Pariatur amet nulla nostrud consequat laboris ipsum excepteur aliqua Lorem excepteur. Adipisicing eu laboris culpa nulla nisi nostrud nulla Lorem. Duis aliquip aliqua elit anim anim dolore ipsum cupidatat sit amet voluptate nostrud. Eiusmod officia sunt elit ad. Ipsum nulla excepteur consequat fugiat cillum laboris tempor labore sit aliqua do ad.\r\n",
    "registered": "2014-07-20T10:02:48 -01:00",
    "latitude": -70.186101,
    "longitude": 86.84117,
    "tags": [
      "proident",
      "Lorem",
      "deserunt",
      "officia",
      "veniam",
      "Lorem",
      "nulla"
    ],
    "greeting": "Hello, Kent Newton! You have 2 unread messages.",
    "favoriteFruit": "apple"
  },
  {
    "id": "5886141786da15797e9f72c2",
    "index": 3,
    "guid": "a7f0cda3-1821-480b-b512-a24edf40d4c1",
    "isActive": true,
    "balance": "$2,735.72",
    "picture": "http://placehold.it/32x32",
    "age": 32,
    "eyeColor": "green",
    "name": "Lenore Russell",
    "gender": "female",
    "company": "PLASMOSIS",
    "email": "lenorerussell@plasmosis.com",
    "phone": "+1 (993) 593-3820",
    "address": "295 Kossuth Place, Leyner, Arkansas, 1730",
    "about": "Elit exercitation commodo in reprehenderit elit Lorem cillum aliquip. Dolor sit reprehenderit ullamco nostrud et est esse minim pariatur cupidatat duis. Ea amet veniam laborum pariatur consequat amet amet voluptate id aute. Est esse Lorem tempor dolore sunt. Aliquip aliquip qui ad ad exercitation sit cillum id reprehenderit mollit ipsum qui ut aliquip. Sit eu laboris amet occaecat elit et.\r\n",
    "registered": "2016-03-27T02:14:38 -01:00",
    "latitude": 8.990375,
    "longitude": -57.752703,
    "tags": [
      "et",
      "in",
      "tempor",
      "enim",
      "adipisicing",
      "aliqua",
      "fugiat"
    ],
    "greeting": "Hello, Lenore Russell! You have 3 unread messages.",
    "favoriteFruit": "apple"
  },
  {
    "id": "58861417a1d7c192e0324805",
    "index": 4,
    "guid": "ee80a07d-bd53-4234-9065-f2df252346ab",
    "isActive": false,
    "balance": "$2,380.42",
    "picture": "http://placehold.it/32x32",
    "age": 35,
    "eyeColor": "brown",
    "name": "Mosley Hendrix",
    "gender": "male",
    "company": "PHUEL",
    "email": "mosleyhendrix@phuel.com",
    "phone": "+1 (825) 437-2089",
    "address": "999 Pine Street, Tyro, Nebraska, 315",
    "about": "Minim occaecat ex occaecat id ex ut velit deserunt ut voluptate. Aliquip est aliquip ad labore deserunt sunt in nostrud et. Reprehenderit eu ea non officia laboris non. Minim enim irure ullamco proident minim incididunt sint voluptate.\r\n",
    "registered": "2016-07-08T02:37:35 -01:00",
    "latitude": 16.296551,
    "longitude": -6.23801,
    "tags": [
      "sit",
      "qui",
      "voluptate",
      "sit",
      "aliqua",
      "ipsum",
      "sunt"
    ],
    "greeting": "Hello, Mosley Hendrix! You have 5 unread messages.",
    "favoriteFruit": "apple"
  },
  {
    "id": "58861417578cbec8ed4c5013",
    "index": 5,
    "guid": "1185c703-99ec-4edb-b15f-8b604ea126c5",
    "isActive": true,
    "balance": "$3,184.79",
    "picture": "http://placehold.it/32x32",
    "age": 28,
    "eyeColor": "blue",
    "name": "Olivia Mccall",
    "gender": "female",
    "company": "HONOTRON",
    "email": "oliviamccall@honotron.com",
    "phone": "+1 (967) 492-2427",
    "address": "133 Autumn Avenue, Oretta, Federated States Of Micronesia, 2125",
    "about": "Qui nostrud ipsum culpa nisi. Ipsum id consectetur elit nulla. Tempor quis ea tempor proident. Aute commodo est esse ea qui enim ipsum sunt aute labore proident minim ipsum exercitation.\r\n",
    "registered": "2016-08-04T10:40:52 -01:00",
    "latitude": -87.524594,
    "longitude": 21.696993,
    "tags": [
      "qui",
      "esse",
      "sit",
      "consectetur",
      "ea",
      "nulla",
      "dolore"
    ],
    "greeting": "Hello, Olivia Mccall! You have 10 unread messages.",
    "favoriteFruit": "apple"
  }
];


export default SearchFixtures
