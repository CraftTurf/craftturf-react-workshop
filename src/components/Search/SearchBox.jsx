import * as React from 'react';
import SearchFixtures from './Search.fixtures'

const SearchBox = ({
  onChange,
   value,
    list,
  onItemClick
 }) => {
  return(
    <div>
    <input
      type='text'
      onChange={onChange}
      value={value}
    />
    { value.length > 0 &&
      list.map(all => {
      const { name, id } = all;
      return (
        <div
          onClick={(e) => onItemClick(e, name)}
          key={id}
          >
          <p>{name}</p>
        </div>
      );
    })}
  </div>
  );
}

SearchBox.PropTypes = {
  list: React.PropTypes.object,
  value: ''
}

export default SearchBox;
