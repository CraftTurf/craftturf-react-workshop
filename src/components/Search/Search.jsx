import * as React from 'react';
import SearchBox from './SearchBox'

export default class Search extends React.Component
{
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    this.handleItemClick = this.handleItemClick.bind(this);

    this.state = {
      value: ''
    };
  }

  handleChange(e) {
    this.setState({
      value: e.target.value
    });
  }

  handleItemClick(e, i) {
    this.setState({
      value: i
    });
    e.preventDefault();
  }

  render() {
    let search = this.state.value.toLowerCase();
    let filteredItems;

    if(this.state.value.length > 0) {
      filteredItems = this.props.list.filter((l) => {
        let allSearch = l.name.toLowerCase().match(search);
        return allSearch;
      })
    }
    return(
      <SearchBox
        onChange={this.handleChange}
        list={filteredItems}
        value={this.state.value}
        onItemClick={this.handleItemClick}
      />
    );
  }

  handleChange(e) {
    this.setState({
      value: e.target.value
    });

  }
}

Search.defaultProps = {
  list: {},
}

Search.PropTypes = {
  onItemClick: React.PropTypes.func
}
