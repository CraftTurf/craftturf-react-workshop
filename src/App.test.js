import React from 'react';
import renderer from 'react-test-renderer';
import MainCounter from './components/Counter/MainCounter';
import Search from './components/Search/Search';
import SearchFixtures from './components/Search/Search.fixtures';

describe('<MainCounter/>', () => {
  it(' should render Snapshot of MainCounter', () => {
    const component = renderer.create(
      <MainCounter />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onClick;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.children;
    tree = components.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Search/>', () => {
  it(' should render Snapshot', () => {
    const component = renderer.create(
      <Saerch list={SearchFixtures}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.handleChange;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.handleItemClick;
    tree = components.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
