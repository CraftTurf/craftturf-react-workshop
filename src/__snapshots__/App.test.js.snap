exports[`<MainCounter/>  should render Snapshot 1`] = `
<div>
  <p>
    0
  </p>
  <button
    className="button_1od8frj"
    onClick={[Function]}>
    Increment
  </button>
  <button
    className="button_1od8frj"
    onClick={[Function]}>
    Decrement
  </button>
</div>
`;

exports[`<MainCounter/>  should render Snapshot 2`] = `
<div>
  <p>
    0
  </p>
  <button
    className="button_1od8frj"
    onClick={[Function]}>
    Increment
  </button>
  <button
    className="button_1od8frj"
    onClick={[Function]}>
    Decrement
  </button>
</div>
`;

exports[`<MainCounter/>  should render Snapshot of MainCounter 1`] = `
<div>
  <p>
    0
  </p>
  <button
    className="button_1m86zok"
    onClick={[Function]}>
    Increment
  </button>
  <button
    className="button_1m86zok"
    onClick={[Function]}>
    Decrement
  </button>
</div>
`;

exports[`<MainCounter/>  should render Snapshot of MainCounter 2`] = `
<div>
  <p>
    0
  </p>
  <button
    className="button_1m86zok"
    onClick={[Function]}>
    Increment
  </button>
  <button
    className="button_1m86zok"
    onClick={[Function]}>
    Decrement
  </button>
</div>
`;
